File: src/tree/N_aryTreePostorderTraversal.java
package tree;

import java.util.ArrayList;
import java.util.List;

public class N_aryTreePostorderTraversal {

    public static void main(String[] args) {

    }

    public List<Integer> postorder(Node root) {
        List<Integer> postList = new ArrayList<Integer>();

        ordering(root, postList);


        return postList;
    }

    public void ordering(Node current, List<Integer> ans) {

        if(current==null){
            return;
        }

        for(Node child: current.children){
            ordering(child, ans);
        }

        ans.add(current.val);
    }

    class Node {
        public int val;
        public List<Node> children;

        public Node() {
        }

        public Node(int _val) {
            val = _val;
        }

        public Node(int _val, List<Node> _children) {
            val = _val;
            children = _children;
        }
    }

    ;
}

File: bin/tree/js/N_aryTreePostorderTraversal.js
function Node(val, children) {
    this.val = val;
    this.children = children;
}

/**
 * @param {Node} root
 * @return {number[]}
 */
var postorder = function (root) {
    let ans = [];
    let stack = [];

    if (root) {
        stack.push(root);
    }

    while (stack.length !== 0) {
        let current = stack.pop();
        ans.push(current.val);

        if (current.children) {
            current.children.forEach((node) => {
                stack.push(node);
            });
        }
    }

    return ans.reverse()
};